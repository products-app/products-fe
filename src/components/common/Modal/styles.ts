import { styled } from '@lebernardo/react'

export const CloseButton = styled('button', {
  display: 'flex',
  justifyContent: 'flex-end',
  padding: '.5rem',
  width: '24rem',
})
