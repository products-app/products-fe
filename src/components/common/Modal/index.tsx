'use client'
import Modal from 'react-modal'
import { X as CloseIcon } from 'phosphor-react'
import { CloseButton } from './styles'

const customStyles = {
  content: {
    top: '50%',
    left: '50%',
    right: 'auto',
    bottom: 'auto',
    marginRight: '-50%',
    transform: 'translate(-50%, -50%)',
    backgroundColor: '#e1e1e1',
  },
  overlay: {
    backgroundColor: 'rgba(0,0,0,.7)',
    zIndex: '9999',
  },
}

Modal.setAppElement('body')

type ModalContainerProps = {
  children: React.ReactNode
  open: boolean
  setOpen: (open: boolean) => void
}

const ModalContainer = ({ children, open, setOpen }: ModalContainerProps) => {
  const closeModal = () => {
    setOpen(false)
  }

  return (
    <div>
      <Modal
        isOpen={open || false}
        onRequestClose={closeModal}
        style={customStyles}
        contentLabel="Checkout"
      >
        <CloseButton aria-label="close-modal" onClick={closeModal}>
          <CloseIcon />
        </CloseButton>

        {children}
      </Modal>
    </div>
  )
}

const defaultProps = {
  children: <></>,
  open: false,
}

ModalContainer.defaultProps = defaultProps
export default ModalContainer
