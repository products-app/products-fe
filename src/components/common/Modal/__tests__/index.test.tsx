import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import Modal from '../index'
import React from 'react'

const props = {
  open: true,
  setOpen: jest.fn(),
}

describe('Modal component', () => {
  it('should render correctly', () => {
    const { baseElement } = render(<Modal {...props} />)
    expect(baseElement).toMatchSnapshot()
  })

  it('should click the close button correctly', async () => {
    render(
      <Modal {...props}>
        <div>test</div>
      </Modal>,
    )

    expect(screen.getByText('test')).toBeInTheDocument()
    const modalWindow = screen.getByRole('dialog', { hidden: true })
    expect(modalWindow).toBeInTheDocument()

    const handleClick = jest.fn()
    const buttonCloseElement = screen.getByRole('button', {
      name: 'close-modal',
      hidden: true,
    })
    buttonCloseElement.addEventListener('click', handleClick, false)

    await userEvent.click(buttonCloseElement)

    expect(handleClick).toHaveBeenCalled()
  })
})
