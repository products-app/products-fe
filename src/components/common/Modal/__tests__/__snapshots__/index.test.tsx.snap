// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component should render correctly 1`] = `
<body
  aria-hidden="true"
  class="ReactModal__Body--open"
>
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div>
    <div />
  </div>
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div
    class="ReactModalPortal"
  >
    <div
      class="ReactModal__Overlay"
      style="position: fixed; top: 0px; left: 0px; right: 0px; bottom: 0px; background-color: rgba(0, 0, 0, 0.7); z-index: 9999;"
    >
      <div
        aria-label="Checkout"
        aria-modal="true"
        class="ReactModal__Content"
        role="dialog"
        style="position: absolute; top: 50%; left: 50%; border: 1px solid #ccc; background: rgb(255, 255, 255); overflow: auto; border-radius: 4px; outline: none; padding: 20px; margin-right: -50%; transform: translate(-50%, -50%); background-color: rgb(225, 225, 225);"
        tabindex="-1"
      >
        <button
          aria-label="close-modal"
          class="c-bzchUs"
        >
          <svg
            fill="currentColor"
            height="1em"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="none"
              height="256"
              width="256"
            />
            <line
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="16"
              x1="200"
              x2="56"
              y1="56"
              y2="200"
            />
            <line
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="16"
              x1="200"
              x2="56"
              y1="200"
              y2="56"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</body>
`;
